modules/standard/IO.chpl

```diff
@@ -2387,7 +2405,7 @@ inline proc channel.commit() where this.locking == false {
                          that the file is not seekable, or that the
                          channel is marked.
  */
-proc channel.seek(start:int, end:int = max(int)) throws {
+proc channel.seek(start:int(64), end:int(64) = max(int(64))) throws {
```

Comment:
I'm not so sure why this change is important because `int` == `int(64)`. Can you share a rationale?

Reply:
I thought `int` was platform-dependent 32-bit or 64-bit, while `int(8)`, `int(16)`, `int(32)` and `int(64)` were guaranteed in size.

Does `int == int(64)` simply because we only target x86-64 and 64-bit GPU processors now, the reasoning being that `int` should always be large enough for any an array size or pointer difference? That seems orthogonal to the file offset issue.

Since this is for file offsets, it doesn't matter whether we're on an ILP32 or LP64 ABI model -- the file offsets should always be 64-bit, even on a ILP32 ABI like x86.

Plus, `int(64)` is used everywhere else offsets are used in the declarations and generated documentation of `IO`, so leaving it out of `channel.seek()` stood out.

Reply:
`int` does not depend on platform (of course it does in C). See https://chapel-lang.org/docs/master/language/spec/types.html#signed-and-unsigned-integral-types

> Plus, int(64) is used everywhere else offsets are used in the declarations and generated documentation of IO, so leaving it out of channel.seek() stood out.

This is an OK rationale